src/app/view/components/_common/site-selector/site-selector.component.ts

```diff
@@ -52,15 +52,26 @@ export class SiteSelectorComponent implements ControlValueAccessor {
 
     constructor(private siteService: SiteService, private config: DotcmsConfig, private iframeOverlayService: IframeOverlayService,
         private paginationService: PaginatorService) {
-
     }
 
     ngOnInit(): void {
         this.paginationService.url = 'v1/site';
-        this.paginateSites();
+
+        this.getSitesList();
 
         this.currentSite = this.siteService.currentSite;
-        this.siteService.switchSite$.subscribe(site => this.currentSite = site);
+
+        this.siteService.switchSite$.subscribe((site) => {
```

Comment:
I think you can do something like this: https://github.com/dotCMS/core-web/pull/383/files#diff-024afe61ad0edf352c55092af19bc243R13 for this.